### PR TITLE
Use getBaseVersion to sync test version across all versions

### PR DESCRIPTION
**Pull request by tyler-cai-microsoft**
[AB#5386](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/5386)

Build-client packages for our internal build pipeline started failing because we started downloading `0.57` -> `0.54` versions. This was because the base version was set as `0.0.0-xyz-test` instead of `2.0.0-dev.1.0.0.xyz-test`

### CODE REVIEW

File: packages/test/test-version-utils/src/baseVersion.ts [hunk 1]
```diff
@@ -3,13 +3,37 @@
...
+	}
+	return pkgVersion;
+}
+
+export const codeVersion = resolveVersion(getCodeVersion(), false);
 export const baseVersion = resolveVersion(
 	(nconf.get("fluid:test:baseVersion") as string) ?? pkgVersion,
 	false,
 );
+export function testBaseVersion(
```
Comment (andre4i) [inline packages/test/test-version-utils/src/baseVersion.ts:30]: Can you add some comments for these exported functions? There was only one function which kind of made sense (getting the base version) but now there are a couple of new concepts introduced (code version, test base version)

File: packages/test/test-version-utils/src/baseVersion.ts [hunk 2]
```diff
@@ -3,13 +3,37 @@
...
  */
 import nconf from "nconf";
+import { isInternalVersionScheme } from "@fluid-tools/version-tools";
 import { resolveVersion } from "./versionUtils.js";
 // This import ensures nconf has been configured to load from correct sources before we compute the right baseVersion.
 // eslint-disable-next-line import/no-unassigned-import
 import "../compatOptions.cjs";
 import { pkgVersion } from "./packageVersion.js";
 
+export function getCodeVersion() {
```
Comment (andre4i) [inline packages/test/test-version-utils/src/baseVersion.ts:13]: Does this need to be exported?

**Review by jatgarg — APPROVED** [review]

**Review by andre4i — APPROVED** [review]
Some minor comments. I think it looks fine.

**Review by tylerbutler — CHANGES_REQUESTED** [review]
Can you add some unit tests?

**Comment by tyler-cai-microsoft** [comment]
> Can you add some unit tests?

added

**Review by tylerbutler — APPROVED** [review]

**Pull request merged**